### PR TITLE
Improves docs on settings

### LINKED ISSUE #2417: Improve the dynaconf/config docs
One thing that's hard is that we don't clearly show example of how to set settings. We should add examples of common settings set.

### PR DESCRIPTION
**Pull request by bmbouter**
This improves (and shortens) the documentation on:

* Applying settings in various ways using dynaconf (with examples).
* Which settings are required

closes #2417

### CODE REVIEW

**Comment by bmbouter** [comment]
@pulp/core can someone review this docs improvement PR please?

File: docs/configuration/applying.rst [hunk 1]
```diff
@@ -3,63 +3,67 @@
...
+starts a Pulp service include::
 
-When using `pulp_installer`
----------------------------
+    Environment="PULP_SETTINGS=/etc/pulp/settings.py"
 
-A Pulp upgrade using `pulp_installer` may override the original ``/etc/pulp/settings.py``.
+A Pulp upgrade using ``pulp_installer`` may override the original ``/etc/pulp/settings.py``.
 To keep your settings through an upgrade, use the `local settings` file located at
 ``/etc/pulp/settings.local.py``.
```
Comment (mdellweg) [inline docs/configuration/applying.rst:69]: Can you add which file take precedence?

File: docs/configuration/applying.rst [hunk 2]
```diff
@@ -3,63 +3,67 @@
...
+The `pulp_installer <https://docs.pulpproject.org/pulp_installer/>`_ enables configuration via a
+settings file that lives at ``/etc/pulp/settings.py``. It does this by having each systemd file that
+starts a Pulp service include::
 
-When using `pulp_installer`
----------------------------
+    Environment="PULP_SETTINGS=/etc/pulp/settings.py"
 
-A Pulp upgrade using `pulp_installer` may override the original ``/etc/pulp/settings.py``.
+A Pulp upgrade using ``pulp_installer`` may override the original ``/etc/pulp/settings.py``.
```
Comment (mdellweg) [inline docs/configuration/applying.rst:67]:
```suggestion
A Pulp upgrade using ``pulp_installer`` will override the original ``/etc/pulp/settings.py``.
```
Can you maybe state unambiguously that this file is "managed" by `pulp_installer`. Do we add a "Don't edit this file!" stanza via the installer to that file?

File: docs/configuration/applying.rst [hunk 3]
```diff
@@ -3,63 +3,67 @@
...
 
-This file should have permissions of:
+    $ cat /etc/pulp/settings.py
+    SECRET_KEY="This should be a 50 chars or longer unique secret!"
 
-* mode: 640
-* owner: root
-* group: pulp (the group of the account that pulp runs under)
-* SELinux context: system_u:object_r:etc_t:s0
+This settings file ends in .py so it needs to be valid Python (so single quotes are necessary), but
```
Comment (mdellweg) [inline docs/configuration/applying.rst:44]:
```suggestion
In this example the settings file ends in ".py" so it needs to be valid Python, but
```
Boolean values or numbers should not be quoted... And double quotes are fine too.

File: docs/configuration/applying.rst [hunk 4]
```diff
@@ -3,63 +3,67 @@
...
+Environment Variables
 ---------------------
 
-To configure Pulp by settings file, you must set the format and location of the config file by
-specifying the ``PULP_SETTINGS`` environment variable. For example, if you wanted to use Python to
-specify your configuration, and provide it at ``/etc/pulp/settings.py`` you could::
+Configuration by specifying environment variables is enabled by default. Any
+:ref:`Setting <settings>` can be configured using Dynaconf by prepending ``PULP_`` to the setting
+name. For example :ref:`SECRET_KEY <secret-key-setting>` can be specified by exporting the
+``PULP_SECRET_KEY`` variable, for example::
```
Comment (mdellweg) [inline docs/configuration/applying.rst:23]: [Maybe too pedantic:] Exporting variables is a shell feature. In the end we only rely on providing/setting the variable in the processes environment.

**Comment by bmbouter** [comment]
@mikedep333 Can you also look at this since I'm modifying some info about the installer. You'll notice the removal of some content because I want to document the requirements, versus what I saw here which seemed more like one way to configure it.

File: docs/configuration/applying.rst (hunk 2 above)
Comment (mikedep333) [reply docs/configuration/applying.rst:67]: Yes, the installer puts such a header in the file:
https://github.com/pulp/pulp_installer/blob/main/roles/pulp_common/templates/settings.py.j2#L1

File: docs/configuration/applying.rst (hunk 4 above)
Comment (bmbouter) [reply docs/configuration/applying.rst:23]: It's a subtle difference, but I think it's meaningful. Take a look at the revision here and lmk.

File: docs/configuration/applying.rst [hunk 5]
```diff
@@ -3,63 +3,67 @@
...
 
-* mode: 750
-* owner: root
-* group: pulp (the group of the account that pulp runs under)
-* SELinux context: unconfined_u:object_r:etc_t:s0
+    The configuration file and directories containing the configuration file must be readable by the
+    user Pulp runs as. If using SELinux permissions, assign the following permissions:
 
-An example of a minimal settings.py file is::
+    * The configuration file itself: ``system_u:object_r:etc_t:s0``
```
[the remark this answers is not in the document]
Comment (bmbouter) [reply docs/configuration/applying.rst:52]: Thanks, see revision in next push.

File: docs/configuration/applying.rst (hunk 2 above)
Comment (bmbouter) [reply docs/configuration/applying.rst:67]: I made this change, see next push.

File: docs/configuration/applying.rst (hunk 1 above)
Comment (bmbouter) [reply docs/configuration/applying.rst:69]: I didn't change this because I didn't know. @mikedep333 what takes precedence?

**Comment by bmbouter** [comment]
@mdellweg @mikedep333 please look once more, I really want to wrap this up so I can take on new work. Thanks!

**Review by ggainey — APPROVED** [review]
I think we're good here, once @mdellweg's question RE precedence is answered!

**Review by dralley — APPROVED** [review]

**Review by mdellweg — APPROVED** [review]
None of the last {ques,sugges}tions are really mandatory.

**Comment by bmbouter** [comment]
I added revised language on the `settings.local.py` taking precedence.

File: docs/configuration/applying.rst [hunk 6]
```diff
@@ -3,63 +3,65 @@
...
----------------------------
+    Environment="PULP_SETTINGS=/etc/pulp/settings.py"
 
-A Pulp upgrade using `pulp_installer` may override the original ``/etc/pulp/settings.py``.
-To keep your settings through an upgrade, use the `local settings` file located at
-``/etc/pulp/settings.local.py``.
+A Pulp upgrade using ``pulp_installer`` will override the original ``/etc/pulp/settings.py``.
+To keep your settings through an upgrade, use the "local settings" file located at
+``/etc/pulp/settings.local.py`` which takes precedence over ``/etc/pulp/settings.py`` on a
+setting-by-setting basis.
```
Comment (mdellweg) [inline docs/configuration/applying.rst:67]: +1

**Review by mdellweg — APPROVED** [review]

**Review by mikedep333 — APPROVED** [review]

**Pull request merged**